controllers/volumereplication_controller.go

```diff
@@ -216,12 +216,7 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		if instance.Status.State != replicationv1alpha1.SecondaryState {
 			replicationErr = r.markVolumeAsSecondary(instance, logger, volumeHandle, parameters, secret)
 			if replicationErr == nil {
-				err := r.updateReplicationStatus(instance, logger, getReplicationState(instance), "volume is marked secondary")
```

Comment:
how will the status state ever change from Primary to Secondary?
is this covered in any test? because after this change, the VR status state doesn't move to Secondary in our tests, but stuck in a `"volume is not ready to use"` loop

Reply:
The call to `updateReplicationStatus` is done after logging, so if there is an issue, it should not be at this point. The status.state is updated as part of `updateReplicationStatus` and that is called in the new code at line 220, right after `markVolumeAsSecondary` returns no errors.

Reply:
@Madhu-1 is testing this out now, and seeing the issue and debugging the same.

Reply:
Fix for it https://github.com/csi-addons/volume-replication-operator/pull/116

Reply:
thanks